app/src/main/java/com/lucas/medicaltools/MainViewState.kt

```diff
@@ -0,0 +1,10 @@
+package com.lucas.medicaltools
+
+import com.lucas.medical_equip.repository.MedicalTool
+
+sealed class MainViewState() {
+    object loadingState: MainViewState()
+    object errorState: MainViewState()
+    data class MedicalToolsState(val medicalTools: List<MedicalTool>?): MainViewState()
```

Comment:
Consolidate to a single state as all the front end needs to know is, hey, here is data, display it. It doesn't care about the logic that go to that point. It only cares that a view state is linked to some code to display the data on the screen.

Reply:
Delete the second state.

Reply:
Re-add your loading state please ;)